qiskit/algorithms/time_evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,53 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Classical Quantum Imaginary Time Evolution."""
+
+from ..time_evolution_problem import TimeEvolutionProblem
+from ..time_evolution_result import TimeEvolutionResult
+from ..imaginary_time_evolver import ImaginaryTimeEvolver
+from .evolve import _evolve
+
+
+class SciPyImaginaryEvolver(ImaginaryTimeEvolver):
+    r"""Classical Evolver for imaginary time evolution.
+
+    Evolves an initial state :math:`|\Psi\rangle` for an imaginary time :math:`\tau = it`
+    under a Hamiltonian :math:`H`, as provided in the ``evolution_problem``.
+    Note that the precision of the evolver does not depend on the number of
+    timesteps taken.
+    """
+
+    def __init__(self, steps: int):
+        r"""
+        Args:
+            steps: The number of timesteps in the simulation.
+        Raises:
+            ValueError: If `steps` is not a positive integer.
+        """
+        if steps <= 0:
```

Comment:
Since `steps` is a public variable this check would have to go into the evolve method (or even better in the common evolve function)